source/jormungandr/jormungandr/scenarios/experimental.py

```diff
@@ -249,85 +408,67 @@ def call_kraken(self, request_type, request, instance, krakens_call):
         if not g.requested_destination:
             g.requested_destination = instance.georef.place(request['destination'])
 
-        for dep_mode, arr_mode in krakens_call:
-            if dep_mode not in g.origins_fallback and request.get('max_duration', 0):
-                g.origins_fallback[dep_mode] = self._get_stop_points(instance,
-                                                                     g.requested_origin, dep_mode,
-                                                                     get_max_fallback_duration(request, dep_mode),
-                                                                     request)
-
-                #Fetch all the stop points of this stop_area and replaces all the durations by 0 in the table
-                #g.origins_fallback[dep_mode]
-                _update_crowfly_duration(instance, g.origins_fallback, dep_mode, request['origin'],
-                                         crow_fly_stop_points)
-
-            if arr_mode not in g.destinations_fallback and request.get('max_duration', 0):
-               g.destinations_fallback[arr_mode] = self._get_stop_points(instance,
-                                                                         g.requested_destination, arr_mode,
-                                                                         get_max_fallback_duration(request, arr_mode),
-                                                                         request,
-                                                                         reverse=True)
-
-               #Fetch all the stop points of this stop_area and replaces all the durations by 0 in the table
-               #g.destinations_fallback[arr_mode]
-               _update_crowfly_duration(instance, g.destinations_fallback, arr_mode, request['destination'],
-                                        crow_fly_stop_points)
+        worker = Worker(instance, krakens_call, request)
+
+        if request.get('max_duration', 0):
+            futures = worker._get_crowfly_futures(g.requested_origin,
+                                            g.requested_destination, g.origins_places_crowfly,
+                                            g.destinations_places_crowfly, 5000)
+            gevent.joinall(futures)
+
+            futures = worker._get_routing_matrix_fustures(g.requested_origin, g.requested_destination,
```

Comment:
There is a lot of abstraction level and absolute not explanation about what happen, why at that moment.
I have count at least seven steps, why do we do one before the other?